js/src/components/DirectedRectEdge.vue

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+  import Vue, { ComponentOptions } from "vue";
+  import Component from "vue-class-component";
+  import { Prop } from "vue-property-decorator";
+  import BaseEdge from "./BaseEdge.vue";
+
+  /**
+   * An edge with an arrow pointing to the target, with support for text.
+   */
+  @Component
+  export default class DirectedRectEdge extends BaseEdge {
+
+    /** The x-coordinate to place the start of the path. It is adjusted to be on the edge of the node. */
+    get adjustedX1() {
+      return this.pathLength === 0 ? 0 : this.x1;
+    }
+
+    /** The y-coordinate to place the start of the path. It is adjusted to be on the edge of the node. */
+    get adjustedY1() {
+      if (this.pathLength === 0) return 0;
+      return this.y1;
+    }
+
+    /** The x-coordinate to place the end of the path. It is adjusted to be on the edge of the node. */
+    get adjustedX2() {
+      let point = this.intersectPoint();
+      return point ? point.x : this.x2;
+    }
+
+    /** The y-coordinate to place the end of the path. It is adjusted to be on the edge of the node. */
+    get adjustedY2() {
+      let point = this.intersectPoint();
+      return point ? point.y : this.y2;
+    }
+
+    slope() : number {
+      if(this.deltaX === 0){
+        return this.deltaY / 0.00001;
```

Comment:
I think we should add a comment explaining this hack, and perhaps define the 0.00001 as a constant.